### PR TITLE
Regenerating auto-gen clients - builder parameterization

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
Depends on https://github.com/googleapis/gax-java/pull/414

### CODE REVIEW

File: google-cloud-logging/src/main/java/com/google/cloud/logging/v2/MetricsSettings.java [hunk 1]
```diff
@@ -176,13 +174,14 @@ public static String getDefaultEndpoint() {
...
   public static InstantiatingGrpcChannelProvider.Builder defaultGrpcTransportProviderBuilder() {
-    return InstantiatingGrpcChannelProvider.newBuilder().setEndpoint(getDefaultEndpoint());
+    return InstantiatingGrpcChannelProvider.newBuilder();
   }
 
   public static TransportChannelProvider defaultTransportChannelProvider() {
     return defaultGrpcTransportProviderBuilder().build();
   }
 
+  @BetaApi("The surface for customizing headers is not stable yet and my change in the future.")
```
Comment (michaelbausor) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/v2/MetricsSettings.java]: s/my/may here and elsewhere

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java [hunk 2]
```diff
@@ -141,6 +141,9 @@ private Publisher(Builder builder) throws IOException {
     if (channelProvider.needsHeaders()) {
       channelProvider = channelProvider.withHeaders(builder.headerProvider.getHeaders());
     }
+    if (channelProvider.needsEndpoint()) {
```
Comment (michaelbausor) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java]: From a review perspective, I think it would be good to put the manual changes in a separate PR, although then it may be impossible to get tests to pass?

File: google-cloud-logging/src/main/java/com/google/cloud/logging/v2/MetricsSettings.java (hunk 1 above)
Comment (garrettjonesgoogle) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/v2/MetricsSettings.java]: This is just embarrassing at this point

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java (hunk 2 above)
Comment (garrettjonesgoogle) [reply google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java]: Correct.

Maybe one process we can follow in the future is to include two commits in the PR - one with only autogen changes, and the next with the manual fixups.
